### PR TITLE
issue fixed #20382

### LINKED ISSUE #20382: View and Edit Cart link not aligned in middle because bellow the link a blank div (class="minicart-widgets") existing has 15px margin top, this div (class="minicart-widgets" ) should be display none or should not come if has no content, should only display if has content
**View and Edit Cart** link not aligned in middle because bellow the link a blank div (**class="minicart-widgets"**) existing has **15px margin top**, this div (**class="minicart-widgets"** ) should be display none or should not come if has no content, should only display if has content.

### Preconditions (*)

1. Magento 2.3
2. php 7.2

### Steps to reproduce (*)

1. Open frontend 
2. Add any product in cart and go to minicart and see **View and Edit Cart** Link  (you will get issue )
3. you will see extra space bellow the link and now inspect you see bank div has 15px margin top.

### Expected result (*)

mini_sol

### Actual result (*)

mini_issue

### PR DESCRIPTION
**Pull request by irajneeshgupta**
### Fixed Issues (if relevant)

1. magento/magento2#20382: View and Edit Cart link not aligned in middle

### Manual testing scenarios (*)

1. Open frontend
2. Add any product in cart and go to minicart and see View and Edit Cart Link
3. You will see extra space bellow the link 

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @irajneeshgupta. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Comment by Karlasa** [comment]
Hi @irajneeshgupta 
Thanks for adding me as a reviewer ;) 
I would not remove styles from that element and even if i would do that, i would fix it in both "Luma" and "Blank" theme. 

Currently i'm not sure where this UI component item is even used? 
example for in default.xml in checkout there are both UI component for promotions and container for blocks. 

Probably UI component is not in use by default and was added for possible add-on modules. 
But anyways my solution would be to add check into html template so i would not render that element if there are no child components to render. 

TLDR: Change in https://github.com/magento/magento2/blob/2.3-develop/app/code/Magento/Checkout/view/frontend/web/template/minicart/content.html#L100
from:
`<div id="minicart-widgets" class="minicart-widgets">`
to: 
`<div id="minicart-widgets" class="minicart-widgets" if="getRegion('promotion').lenght">`

Cheers,

**Review by Karlasa — CHANGES_REQUESTED** [review]
I would instead add check for that component in html so it would not render if there are no content/child elements.

**Comment by irajneeshgupta** [comment]
> I would instead add check for that component in html so it would not render if there are no content/child elements.

Thank you , i am making changes to this commit.

**Comment by irajneeshgupta** [comment]
> Hi @irajneeshgupta
> Thanks for adding me as a reviewer ;)
> I would not remove styles from that element and even if i would do that, i would fix it in both "Luma" and "Blank" theme.
> 
> Currently i'm not sure where this UI component item is even used?
> example for in default.xml in checkout there are both UI component for promotions and container for blocks.
> 
> Probably UI component is not in use by default and was added for possible add-on modules.
> But anyways my solution would be to add check into html template so i would not render that element if there are no child components to render.
> 
> TLDR: Change in https://github.com/magento/magento2/blob/2.3-develop/app/code/Magento/Checkout/view/frontend/web/template/minicart/content.html#L100
> from:
> `<div id="minicart-widgets" class="minicart-widgets">`
> to:
> `<div id="minicart-widgets" class="minicart-widgets" if="getRegion('promotion').lenght">`
> 
> Cheers,

Hi @Karlasa 
Please review, i have updated my commits.
Thank you.

**Review by Karlasa — APPROVED** [review]

**Review by sivaschenko — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko, thank you for the review.
[ENGCOM-4066](https://jira.corp.magento.com/browse/ENGCOM-4066) has been created to process this Pull Request
